src/coreclr/jit/regalloc.cpp

```diff
@@ -342,18 +333,7 @@ void Compiler::raMarkStkVars()
 
             if (opts.compDbgCode && !stkFixedArgInVarArgs && lclNum < info.compLocalsCount)
             {
-                if (varDsc->lvRefCnt() == 0)
-                {
-                    assert(!"unreferenced local in debug codegen");
-                    varDsc->lvImplicitlyReferenced = 1;
-                }
-
-                needSlot |= true;
-
-                if (!varDsc->lvIsParam)
-                {
-                    varDsc->lvMustInit = true;
-                }
+                assert(!"unreferenced local in debug codegen");
```

Comment:
`needSlot` is always `false` now; you could remove it entirely.

Reply:
Ah, good catch

Reply:
I modernized the styling of the whole file and removed the outdated comment at the top as well.

Reply:
I also removed the assert here as it doesn't seem that valuable, but let me know if you disagree.